core/chains/evm/client/simulated_backend_client.go

```diff
@@ -111,6 +111,10 @@ func (c *SimulatedBackendClient) CallContext(ctx context.Context, result interfa
 	}
 }
 
+func (c *SimulatedBackendClient) FilterEvents(ctx context.Context, q ethereum.FilterQuery) (logs []types.Log, err error) {
+	return c.b.FilterLogs(ctx, q)
+}
+
```

Comment:
If i remember correctly, we are trying to use the more generic naming `Events` rather than `Logs`

so in the generic interface there is only `FilterEvents` so we wrap `FilterLogs` into `FilterEvents` to conform to the interface

Reply:
Yep, I tried shifting to the `FilterEvents` same as we did with nonce -> sequence across the code base, but it's actually too large of a change for this PR, so I limited it to only conforming to the generic interface for now.